ourjapanlife/findadoc-server
# 27 prisma orm

Resolves #27 

Since I've been blocked on the docker-compose bit in #36 (and I am waiting on help from the Prisma slack) I thought I'd start working on the postgres / prisma ORM here. Adding docker will be another PR (though should make postgres setup much easier than bare-metal)

- [x] add prisma
- [x]  postgres setup script & instructions
- [x] prisma.schema file 
- [x] migrations file
- [x] instructions for how to use prisma
- 🏉  seed data for basics like languages, degrees, etc. => move to another issue #52 

## Notes

I started with our [DataModel document](https://www.notion.so/ourjapanlife/Data-Model-85db33389bf341f59d0ea5fa2d4b759c), which has a ton of feedback on it. I drew a model in Miro of what I think makes the most sense for the DB at this point. Please check it out here: https://miro.com/app/board/uXjVP3xzRlk=/

## Testing instructions

1. Follow the steps in the readme regarding DB setup to verify that a. the readme is correct and b. it works

2. check the migration file `prisma/migrations/ **` and ensure the generated SQL makes sense

3. You can also try prisma studio: `yarn prisma studio` but there's no data yet
